### PR TITLE
DXE-2161 Update from v3 to  v4

### PR DESCRIPTION
**Pull request by Slonimskaia**
* Updated documentation 
* Created v3 branch from [commit](https://github.com/akamai/AkamaiOPEN-edgegrid-golang/commit/756dbaa8efa3298fda15cc41ded8f2dfb7a6f315) which is the latest commit for the v3.1.0 release

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
     branches:
       - v1
       - v2
+      - v3
```
Comment (mgwoj) [inline .github/workflows/checks.yml:10]: Not needed as there is no `v3` branch. There is only `v3` version

**Review by mgwoj — CHANGES_REQUESTED** [review]

File: .github/workflows/checks.yml (hunk 1 above)
Comment (Slonimskaia) [reply .github/workflows/checks.yml:10]: There is v3 branch, I mentioned it in description of this pr

Comment (Slonimskaia) [reply .github/workflows/checks.yml:10]: Do you think that we don't need this v3? If so I can remove it

Comment (mgwoj) [reply .github/workflows/checks.yml:10]: Please keep it

**Pull request merged**